### PR TITLE
Create a hook to set the document title, to replace the old dependency

### PR DESCRIPTION
**Pull request by axbarsan**
Fixes giantswarm/giantswarm#8013

- Remove the 'react-document-title' dependency
- Create a hook to easily set the document title
- Create a component to wrap the setting document title functionality and to mimmick the old interface
- Replace the old 'react-document-title' module to our new one

### CODE REVIEW

**Comment by axbarsan** [comment]
Tests incoming 👍

File: src/components/shared/__tests__/DocumentTitle.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    initialTitle = document.title;
+  });
+
+  afterAll(() => {
+    document.title = initialTitle;
+  });
+
+  beforeEach(() => {
+    document.title = initialTitle;
+  });
```
Comment (ograu) [inline src/components/shared/__tests__/DocumentTitle.js]: nitpick 1: I like to follow a chronological order here, it helps me in understanding what's going on

File: src/components/shared/__tests__/DocumentTitle.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  it('renders without crashing', () => {
+    renderWithProps({});
+  });
+
+  it(`sets title to 'Giant Swarm' if no title prop is provided`, () => {
+    renderWithProps({});
+
+    expect(document.title).toBe('Giant Swarm');
```
Comment (ograu) [inline src/components/shared/__tests__/DocumentTitle.js]: Shouldn't we be using a constant? I've seen `'Giant Swarm'` at least three times spreaded across the codebase

File: src/lib/effects/useDocumentTitle.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  useEffect(() => {
+    const previousTitle = document.title;
+
+    return () => {
+      document.title = previousTitle;
+    };
+  }, []);
+
+  useEffect(() => {
+    document.title = title;
```
Comment (ubergesundheit) [inline src/lib/effects/useDocumentTitle.js]: It seems all occurences of `<DocumentTitle ..></DocumentTitle>` end with ` | Giant Swarm`. Shouldn't we remove it from all components and add it here as a default suffix for all document titles?

Comment (axbarsan) [reply src/lib/effects/useDocumentTitle.js]: This is actually a very good idea 👍

**Pull request merged**